ui/src/warp_runner/commands.rs

```diff
@@ -45,6 +75,36 @@ pub async fn handle_tesseract_cmd(cmd: TesseractCmd, tesseract: &mut Tesseract)
     }
 }
 
+pub async fn handle_raygun_cmd(cmd: RayGunCmd, account: &mut Account, messaging: &mut Messaging) {
+    match cmd {
+        RayGunCmd::InitializeConversations { rsp } => match messaging.list_conversations().await {
+            Ok(convs) => {
+                //println!("warp runner got conversations: {:#?}", convs);
+                let mut all_chats = HashMap::new();
+                for conv in convs {
+                    let chat = conversation_to_chat(conv, account, messaging).await;
+                    all_chats.insert(chat.id, chat);
+                }
+                let _ = rsp.send(Ok(all_chats));
+            }
+            Err(_e) => {
+                // do nothing. will cancel the channel
+            }
+        },
+        RayGunCmd::CreateConversation { recipient, rsp } => {
+            match messaging.create_conversation(&recipient).await {
+                Ok(conv) => {
```

Comment:
```suggestion
                Ok(conv) | Err(Error::ConversationExist {
        conversation: conv,
    }) => {
```

Would suggest trying it in IDE but you should try this error too as a precaution since it would yield a conversation type.